source/main.cc

```diff
@@ -493,48 +493,51 @@ void
 parse_parameters (const std::string &input_as_string,
                   dealii::ParameterHandler  &prm)
 {
-  // try reading on processor 0
-  bool success = true;
-  if (dealii::Utilities::MPI::this_mpi_process (MPI_COMM_WORLD) == 0)
-    try
-      {
-        prm.parse_input_from_string(input_as_string);
-      }
-    catch (const dealii::ExceptionBase &e)
-      {
-        success = false;
-        e.print_info(std::cerr);
-        std::cerr << std::endl;
-      }
```

Comment:
Here, the nonlocality of the `success` variable is what makes the code difficult to read. Moreover, the fact that we catch the exception but then treat it differently than what we do in `main()` is troublesome -- we should have only one way to print errors, and that should happen in `main()`.